lib/spack/spack/package.py

```diff
@@ -1464,7 +1466,40 @@ def do_install(self, **kwargs):
 
     @property
     def test_log_name(self):
-        return 'test-%s-out.txt' % self.spec.format('{name}-{version}-{hash:7}')
+        return 'test-{0}-out.txt' \
+            .format(self.spec.format('{name}-{version}-{hash:7}'))
+
+    @property
+    def test_dir(self):
+        return self.test_stage.join(
+            self.spec.format('{name}-{version}-{hash}'))
+
+    def cache_extra_test_sources(self, srcs):
+        """Copy relative source path(s) to the corresponding install test subdir
+
+        This method is intended as an optional install test setup helper for
+        grabbing source files/directories during the installation process and
+        copying them to the installation test subdirectory for subsequent use
+        during install testing.
+
+        Args:
+            srcs (str or list of str): relative path for files and or
+                subdirectories located in the staged source path that are to
+                be copied to the corresponding location(s) under the install
+                testing directory.
+        """
+        paths = [srcs] if isinstance(srcs, string_types) else srcs
+
+        def skip_file(path):
```

Comment:
NOT A CHANGE REQUEST: for code style, is this better than a lambda function? I ask because I would probably have defaulted to use a lambda function, but maybe that's not as legible? It keeps everything in one place but it does lead to long lines.

Reply:
Changed anyway.